binding/richclient.go

```diff
@@ -0,0 +1,117 @@
+package binding
+
+import (
+	"fmt"
+
+	"github.com/konveyor/controller/pkg/logging"
+	"github.com/konveyor/tackle2-hub/api"
+	"github.com/konveyor/tackle2-hub/settings"
+)
+
+var (
+	Settings = &settings.Settings
+	Log      = logging.WithName("binding")
+)
+
+var (
+	// Addon An addon richClient configured for a task execution.
+	RichClient *Adapter
+	// Client
+	//client *Client
+)
+
+func init() {
+	err := Settings.Load()
+	if err != nil {
+		panic(err)
+	}
+
+	RichClient = newRichClient()
+}
+
+// The RichClient provides API integration.
+type Adapter struct {
+	//	// Task API.
+	//	Task
+	//	// Settings API
+	//	Setting Setting
+	// Application API.
+	Application Application
+	// Identity API.
+	//	Identity Identity
+	//	// Proxy API.
+	//	Proxy Proxy
+	//	// TagCategory API.
+	//	TagCategory TagCategory
+	//	// Tag API.
+	//	Tag Tag
+	//	// File API.
+	//	File File
+	//	// RuleBundle API
+	//	RuleBundle RuleBundle
+	// client A REST client.
+	client *Client
+}
+
+// Client provides the REST client.
+func (h *Adapter) Client() *Client {
+	return h.client
+}
+
+// newRichClient builds a new Addon RichClient object.
+func newRichClient() (richClient *Adapter) {
```

Comment:
Let's rename _newRichClient()_ to just `New(url) (client *RichClient)`.
This will be the _main_ constructor for the package.

Add `RichClient.Login(user, password string) (err error)`

```
client := binding.New("http://localhost:8080/hub")
err := client.Login("elmer", "fudd")
...
```

Reply:
:+1: Updated.